streaming/src/main/scala/org/apache/spark/streaming/receiver/ReactiveReceiver.scala

```diff
@@ -0,0 +1,141 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.streaming.receiver
+
+import org.reactivestreams._
+import org.apache.spark.storage.StorageLevel
+import org.apache.spark.annotation.DeveloperApi
+import scala.collection.mutable.ArrayBuffer
+import scala.util.{Try, Success, Failure}
+import java.lang.IllegalStateException
+
+@DeveloperApi
+abstract class ReactiveReceiver[T](storageLevel: StorageLevel)
```

Comment:
The code looks good, but since this is Developer API you should add some docs, especially on methods that are overridable (`def when*`).